### PR TITLE
Fix missing export files for extensions

### PR DESCRIPTION
**Pull request by D8H**
Fixes https://github.com/4ian/GDevelop/issues/4747 and https://github.com/4ian/GDevelop/issues/745
- https://github.com/4ian/GDevelop/issues/4747
- https://github.com/4ian/GDevelop/issues/745

### CODE REVIEW

File: Core/GDCore/IDE/Events/UsedExtensionsFinder.h [hunk 1]
```diff
@@ -21,18 +21,50 @@ class Behavior;
...
+    return usedExtensions;
+  }
+
+  const std::set<gd::String>& GetUsedIncludeFiles() const {
+    return usedIncludeFiles;
+  }
+
+  const std::set<gd::String>& GetUsedRequiredFiles() const {
+    return usedRequiredFiles;
+  }
```
Comment (4ian) [inline Core/GDCore/IDE/Events/UsedExtensionsFinder.h:36]:
```suggestion
  /**
   * The extensions used by the project (or part of it).
   */
  const std::set<gd::String>& GetUsedExtensions() const {
    return usedExtensions;
  }
  
  /**
   * The include files used at runtime by the project (or part of it).
   */
  const std::set<gd::String>& GetUsedIncludeFiles() const {
    return usedIncludeFiles;
  }
  
  /**
   * The additional files required at runtime by the project (or part of it).
   */
  const std::set<gd::String>& GetUsedRequiredFiles() const {
    return usedRequiredFiles;
  }
```

File: Core/GDCore/IDE/Events/UsedExtensionsFinder.h [hunk 2]
```diff
@@ -21,18 +21,50 @@ class Behavior;
...
+
+  std::set<gd::String>& GetUsedExtensions() {
```
Comment (4ian) [inline Core/GDCore/IDE/Events/UsedExtensionsFinder.h:38]: (+ same comments here)

**Review by 4ian — APPROVED** [review]
Overall, lgtm!

About #745, does this solve the entire problem? Are we now always 100% correct related to the extensions being included (I believe so).
Are we also 100% correct about an extension that would use another one (even if there is no dependency between extensions being installed, I believe this created issue in the past if you try to still make on your own an extension that was calling another that was calling another).

**Comment by D8H** [comment]
I tested: behavior doStepPreEvent --> free function --> other extension B free function --> other extension C free function --> log
https://www.dropbox.com/s/m3jev75nwtaivq7/IncludeTest.zip?dl=1

**Comment by Bouh** [comment]
> About https://github.com/4ian/GDevelop/issues/745, does this solve the entire problem? Are we now always 100% correct related to the extensions being included (I believe so).

> I tested: behavior doStepPreEvent --> free function --> other extension B free function --> other extension C free function --> log https://www.dropbox.com/s/m3jev75nwtaivq7/IncludeTest.zip?dl=1

So you mean yes?
Can we close the issue ?

**Pull request merged**